kernel/aster-nix/src/fs/ext2/inode.rs

```diff
@@ -672,17 +672,19 @@ impl Inner {
         };
         self.page_cache.discard_range(offset..offset + read_len);
 
-        let mut buf_offset = 0;
-        for bid in Bid::from_offset(offset)..Bid::from_offset(offset + read_len) {
-            let frame = FrameAllocOptions::new(1)
-                .uninit(true)
-                .alloc_single()
-                .unwrap();
+        let buf_nblocks = buf.len() / BLOCK_SIZE;
+        let start_bid = Bid::from_offset(offset).to_raw() as Ext2Bid;
+        // Read multi blocks in batches
+        if buf_nblocks > 1 {
             self.inode_impl
-                .read_block_sync(bid.to_raw() as Ext2Bid, &frame)?;
-            frame.read_bytes(0, &mut buf[buf_offset..buf_offset + BLOCK_SIZE])?;
-            buf_offset += BLOCK_SIZE;
+                .read_blocks_sync(start_bid, buf_nblocks, buf)?;
```

Comment:
The second argument `buf_nblocks` can be calculated with `buf.len() / BLOCK_SIZE`, so it is useless.

Reply:
My initial intention is to distinguish tons of API like write_***(&self, offset: usize, buf: &[u8]), so I shape that to pub fn write_blocks_sync(&self, start_bid: Ext2Bid, nblocks: usize, raw_buf: &[u8])to make a apparent difference.